cxx/clean_relation.hh

```diff
@@ -26,7 +27,7 @@ class T_clean_relation {
   // TODO(emilyaf): Enable observed vs. latent.
   // bool is_observed;
 
-  DistributionSpec distribution_spec;
+  DistributionSpec spec;
```

Comment:
Slight preference for keeping this as distribution_spec or dist_spec because we now have a lot of different specs floating around.

Reply:
I did this so I can `std::visit` a `std::variant<T_clean_relation, T_noisy_relation>` with a single function that access `spec` (which is either a DistributionSpec or an EmissionSpec). WDYT of naming them both `cluster_spec` instead?

Reply:
Okay.  In that case, either spec or cluster_spec is fine, but please add a comment describing how the name duplication is load-bearing.

Reply:
I rethought this and changed it back to `distribution_spec` and `emission_spec`.